### PR TITLE
Making doozer installable

### PR DESCRIPTION
**Pull request by adammhaile**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,6 +9,11 @@ dist
...
 .eggs
 .cache
+venv
+cover
+*.egg-info
+
+
 
 # Ignore emacs backup files
 *~
```
Comment (tbielawa) [inline .gitignore]: I like this line the most.

File: doozer [hunk 2]
```diff
@@ -31,24 +35,22 @@
...
 @click.option("--branch", default=None, metavar='BRANCH',
-              help="Branch to override any default in group.yml.")
+              help="DistGit to override any default in group.yml.")
 @click.option('--stage', default=False, is_flag=True, help='Force checkout stage branch for sources in group.yml.')
 @click.option("-i", "--images", default=[], metavar='NAME', multiple=True,
               help="Name of group image member to include in operation (all by default). Can be comma delimited list.")
 @click.option("-r", "--rpms", default=[], metavar='NAME', multiple=True,
               help="Name of group rpm member to include in operation (all by default). Can be comma delimited list.")
-@click.option('--wip', default=False, is_flag=True, help='Load WIP RPMs/Images in addition to those specified, if any')
+@click.option('--load-wip', default=False, is_flag=True, help='Load WIP RPMs/Images in addition to those specified, if any')
```
Comment (tbielawa) [inline doozer]: Solid change. Way clearer.

File: doozer [hunk 3]
```diff
@@ -31,24 +35,22 @@
...
-              help="Username for rhpkg. Env var: OIT_USER")
+@click.option("--data-path", metavar='PATH', default=None,
+              help="Git repo or directory containing groups metadata")
+@click.option("--working-dir", metavar='PATH', default=None,
+              help="Existing directory in which file operations should be performed.\n Env var: DOOZER_WORKING_DIR")
+@click.option("--user", metavar='USERNAME', default=None,
+              help="Username for rhpkg. Env var: DOOZER_USER")
 @click.option("-g", "--group", default=None, metavar='NAME',
-              help="The group of images on which to operate.")
+              help="The group of images on which to operate. Env var: DOOZER_GROUP")
```
Comment (tbielawa) [inline doozer]: Is there suppoesd to be an `envvar` parameter here?

File: doozer [hunk 4]
```diff
@@ -410,15 +413,15 @@ def images_foreach(runtime, cmd, message, push):
...
-    oit_distgit_key : The name of the distgit_key used with -i, -x for this image
-    oit_image_name : The name of the image from Dockerfile
-    oit_image_version : The current version found in the Dockerfile
-    oit_group: The group for this invocation
-    oit_metadata_dir: The directory containing the oit metadata
-    oit_working_dir: The current working directory
+    doozer_repo_name : The name of the distgit repository
+    doozer_repo_namespace : The distgit repository namespaces (e.g. containers, rpms))
+    doozer_config_filename : The config yaml (basename, no path) associated with an image
+    doozer_distgit_key : The name of the distgit_key used with -i, -x for this image
```
Comment (tbielawa) [inline doozer]: Are these all actually lowercase, or uppercase?

File: doozerlib/runtime.py [hunk 5]
```diff
@@ -851,14 +851,19 @@ def resolve_metadata(self):
...
         """
 
-        if self.metadata_dir is None:
-            self.metadata_dir = constants.OCP_BUILD_DATA_RW
+        if self.data_path is None:
+            raise DoozerFatalError(
+                ("No metadata path provided. Must be set via one of:\n"
+                 "* data_path key in {}\n"
+                 "* doozer --datapath [PATH|URL]\n"
+                 "* Environment variable DOOZER_DATA_PATH\n"
```
Comment (tbielawa) [inline doozerlib/runtime.py]: Yeah, i pointed out in an earlier comment about upper/lowercase vars. Here it's upper.

File: doozerlib/util.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import click
+
+
```
Comment (tbielawa) [inline doozerlib/util.py]: love it

File: doozer (hunk 4 above)
Comment (adammhaile) [reply doozer]: lowercase, those used to be `oit_*`, I just did a find/replace in that section. Honestly that whole foreach command is in dire need of an overhaul. I'm not 100% sure it works as intended anymore. It's very old.

File: doozerlib/runtime.py (hunk 5 above)
Comment (adammhaile) [reply doozerlib/runtime.py]: That is an actual environment variable, the ones above are text replacement options for the foreach command. Very diff things.

**Comment by tbielawa** [comment]
Looks good to me. @sosiouxme were you writing up a review too?

**Comment by adammhaile** [comment]
@tbielawa @sosiouxme Note. After this merges I will take care of making that released branch and install it on buildvm from there so I can start working on moving the Jenkins jobs over.
However, at this point I have a feeling that work will have a PR by end of tomorrow but probably not merge until after the holiday. It's one thing to merge new code on a Friday, it's another to do it on a Friday before being gone for a week ;)

**Pull request merged**